api/api/serializers/image_serializers.py

```diff
@@ -118,13 +119,35 @@ class Meta:
 class OembedRequestSerializer(serializers.Serializer):
     """Parse and validate oEmbed parameters."""
 
-    url = serializers.CharField(
+    url = serializers.URLField(
+        allow_blank=False,
         help_text="The link to an image present in Openverse.",
     )
 
-    @staticmethod
-    def validate_url(value):
-        return add_protocol(value)
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+
+        url = data["url"]
+        if url.endswith("/"):
+            url = url[:-1]
+        identifier = url.rsplit("/", 1)[1]
+
+        try:
+            uuid = UUID(identifier)
+        except ValueError:
+            raise serializers.ValidationError(
+                {"Could not parse identifier from URL.": data["url"]}
+            )
+
+        try:
+            image = Image.objects.get(identifier=uuid)
+        except (Image.DoesNotExist, ValidationError):
+            raise serializers.ValidationError(
+                {"Could not find image from the provided URL": data["url"]}
+            )
```

Comment:
IIRC in Python, try/excepts can be a little expensive. We could potentially combine these to make it a bit simpler:
```suggestion
        try:
            uuid = UUID(identifier)
            image = Image.objects.get(identifier=uuid)
        except ValueError:
            raise serializers.ValidationError(
                {"Could not parse identifier from URL.": data["url"]}
            )
        except (Image.DoesNotExist, ValidationError):
            raise serializers.ValidationError(
                {"Could not find image from the provided URL": data["url"]}
            )
```